exotica/src/Scene.cpp

```diff
@@ -689,6 +689,15 @@ void Scene::addObject(const std::string& name, const KDL::Frame& transform, cons
     if (updateCollisionScene) updateCollisionObjects();
 }
 
+void Scene::addObjectToEnvironment(const std::string& name, const KDL::Frame& transform, shapes::ShapeConstPtr shape, const bool updateCollisionScene)
+{
+    Eigen::Isometry3d pose;
+    tf::transformKDLToEigen(transform, pose);
+    ps_->getWorldNonConst()->addToObject(name, shape, pose);
```

Comment:
TODO: We should check for uniqueness of the name with all elements in both the scene _and_ the KinematicTree

Reply:
The moveit documentation says about `addToObject`:
> Add a shape to an object.
> If the object already exists, this call will add the shape to the object
> at the specified pose. Otherwise, the object is created and the
> specified shape is added. This calls addToObjectInternal().

So, `addObjectToEnvironment`, hence `addToObject`, will actually compose an object out of shapes that are added to it.

Would it be enought to just check for existing objects in the `KinematicTree`?

Reply:
Yes. I think if we are recreating all of the environment per ``updateSceneframes`` it _should_ be fine. From a brief look, it appears we aren't yet enforcing link names can only be added once in ``KinematicTree`` and this should be fixed in the ``AddElement`` method.

Reply:
Checked with new method `KinematicTree::hasLink`.